grails-app/controllers/org/pih/warehouse/inventory/InventoryItemController.groovy

```diff
@@ -320,14 +320,14 @@ class InventoryItemController {
         requisitionItems = requisitionService.getPendingRequisitionItems(location, product)
         requisitionItems.groupBy { it.requisition }.collect { k, v ->
             itemsMap.put(k, [
+                    picklistItemsByLot: k?.picklist?.picklistItems?.findAll {it.quantity && it.inventoryItem.product == product}?.groupBy {it.inventoryItem.lotNumber},
```

Comment:
I think it may be safer to check if products have the same id

Reply:
Also, that feels like logic that would be useful in the Picklist class itself. 
```
List<PicklistItem> getPicklistItems(Product product) { 
    return picklistItems?.findAll { it.quantity && it.inventoryItem.product?.id == product?.id }
}

def getPicklistItemsByLot(Product product)  { 
    return getPicklistItems(product)?.groupBy {it.inventoryItem.lotNumber}
}
```
then in InventoryItemController
```
itemsMap.put(k, [
    picklistItemsByLot: k?.picklist?.getPicklistItemsGroupByLot(product),
    ...
])
```

Reply:
Done :)